apis/securesourcemanager/v1alpha1/instance.go

```diff
@@ -0,0 +1,126 @@
+
+import (
+	"reflect"
+
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// SchemeBuilder is used to add go types to the GroupVersionKind scheme.
+	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
+
+	// AddToScheme is a global function that registers this API group & version to a scheme
+	AddToScheme = SchemeBuilder.AddToScheme
+
+	SecureSourceManagerInstanceGVK = schema.GroupVersionKind{
+		Group:   SchemeGroupVersion.Group,
+		Version: SchemeGroupVersion.Version,
+		Kind:    reflect.TypeOf(SecureSourceManagerInstance{}).Name(),
+	}
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories=gcp,shortName=gcpsecuresourcemanagerinstance;gcpsecuresourcemanagerinstances
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/stability-level=alpha";"cnrm.cloud.google.com/system=true"
+// +kubebuilder:printcolumn:name="Age",JSONPath=".metadata.creationTimestamp",type="date"
+// +kubebuilder:printcolumn:name="Ready",JSONPath=".status.conditions[?(@.type=='Ready')].status",type="string",description="When 'True', the most recent reconcile of the resource succeeded"
+// +kubebuilder:printcolumn:name="Status",JSONPath=".status.conditions[?(@.type=='Ready')].reason",type="string",description="The reason for the value in 'Ready'"
+// +kubebuilder:printcolumn:name="Status Age",JSONPath=".status.conditions[?(@.type=='Ready')].lastTransitionTime",type="date",description="The last transition time for the value in 'Status'"
+
+// SecureSourceManagerInstance is the Schema for the monitoring API
+// +k8s:openapi-gen=true
+type SecureSourceManagerInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec SecureSourceManagerInstanceSpec `json:"spec,omitempty"`
+
+	Status SecureSourceManagerInstanceStatus `json:"status,omitempty"`
+}
+
+// +kcc:proto=google.cloud.securesourcemanager.v1.Instance
+type SecureSourceManagerInstanceSpec struct {
+	/* Immutable. The Project that this resource belongs to. */
+	ProjectRef refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Location of the instance. */
+	Location string `json:"location"`
+
+	/* Immutable. Optional. The name of the resource. Used for creation and acquisition. When unset, the value of `metadata.name` is used as the default. */
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Optional. Labels as key value pairs.
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Optional. Immutable. Customer-managed encryption key name, in the format
+	//  projects/*/locations/*/keyRings/*/cryptoKeys/*.
```

Comment:
Shall this be a ref? What I'm actually trying to ask is if we shall validate the existence of this kmsKey

Reply:
Yes, it should be a ref!  It's alpha so we can still change it, but ... you're right.  I'll give it a go!